wfl/Database/MongoDBManager.cs

```diff
@@ -19,62 +20,66 @@ public MongoDBManager()
             db = client.GetDatabase("WFL");
         }
 
-        public async Task<Restaurant> InsertRestaurantAsync(RestaurantDto r)
+        private string GenerateCollectionName<T>() where T: new()
         {
-            IMongoCollection<Restaurant> collection = db.GetCollection<Restaurant>("restaurants");
-            Restaurant restaurant = new Restaurant()
+            T t = new T();
```

Comment:
Now I see the requirement for `new ()`...you can do this `Type entityType = typeof(T); string type = entityType.Name;` for what you are trying to do. Again, there are a couple `Name` properties on the type object. So you'll have to find the right one.

- [ ] Remove `new ()` constraint
- [ ] Use `typeof(T)` to get the entity type